### ISSUE TITLE
Threading for fetching Scoresaber and BeatSaver info

### ISSUE THREAD

**Issue by Luux** [Luux/ARBSMapDo#2]
Handling multiple requests at the same time, starting downloads in parallel to searching songs that meet the criteria.

**Comment by Luux** [comment]
Scoresaber doesn't need Threading since we just can set the limit high enough: https://github.com/Luux/ARBSMapDo/commit/7d98d21acd56072c6db5231f40395e4e5b811d9c

**Comment by Luux** [comment]
BeatSaver will be much faster with the local cache (see https://github.com/Luux/ARBSMapDo/issues/6) anyway.

**Comment by Luux** [comment]
I think this is no longer needed for now.
